packages/mobile/src/app/saga.ts

```diff
@@ -145,14 +146,19 @@ function convertQueryToScreenParams(query: string) {
 export function* handleDeepLink(action: OpenDeepLink) {
   const { deepLink, isSecureOrigin } = action
   Logger.debug(TAG, 'Handling deep link', deepLink)
-  const rawParams = parse(deepLink)
+  let link = deepLink
+  if (deepLink.startsWith('https://vlra.app')) {
+    const dynamicLink = yield call([dynamicLinks(), 'resolveLink'], deepLink)
+    link = dynamicLink.url
+  }
```

Comment:
I don't see why we need to manually resolve this.
If the app is opened following a short link, `dynamicLinks().getInitialLink()` would contain the resolved link.

We should check that and also listen to the `onLink` event.
See https://rnfirebase.io/dynamic-links/usage

Reply:
While doing this we need to be careful, there were some subtle issues with handling deeplinks from the various app state (background, foreground, especially on Android).
@etuleu can help you with this too.

Reply:
`dynamicLinks().getInitialLink()` returns the same thing than `dynamicLinks().resolveLink(deepLink)` so we can use that, but listening to the `onLink` event would cause us to handle the link twice since it's already being listened on in `App.tsx` by doing `Linking.addEventListener('url', this.handleOpenURL)`

Reply:
I see, I guess it depends if we want to take advantage of the dynamic link working through a new app install.
Also I thought `onLink` would only be triggered when resolving a dynamic link.